src/Engine/EmitMSIL/Replication.cs

```diff
@@ -492,8 +492,27 @@ private static List<List<ReplicationGuide>> ConstructRepGuides(string[][] replic
 
             var replicationTrials = Replicator.BuildReplicationCombinations(instructions, arguments);
 
-            //TODO_MSIL: implement case 2
             #region Case 2: Replication and replication guide with exact match
+            {
+                //Build the possible ways in which we might replicate
+                foreach (List<ReplicationInstruction> replicationOption in replicationTrials)
+                {
+                    List<List<CLRStackValue>> reducedParams = Replicator.ComputeAllReducedParams(arguments, replicationOption, runtimeCore);
+                    HashSet<CLRFunctionEndPoint> lookups;
+                    if (FunctionGroup.CanGetExactMatchStatics(reducedParams, funcGroup, runtimeCore, out lookups))
+                    {
+                        if (replicationInstructions == null || CallSite.IsSimilarOptionButOfHigherRank(replicationInstructions, replicationOption))
+                        {
+                            // We have a cluster of FEPs that can be used to dispatch the array
+                            resolvedFeps = new List<CLRFunctionEndPoint>(lookups);
+                            replicationInstructions = replicationOption;
+                            matchFound = true;
```

Comment:
@pinzart90 should this break?

Reply:
Looks like it could. But then it would be different than the old VM